### ISSUE TITLE
Display cluster labels in the UI

### ISSUE THREAD

**Issue by agagniere** [argoproj/argo-cd#10997]
# Summary

## Feature proposed 

Adding a list of labels and annotations in the cluster details view, displaying the metadata of the argo-cd Cluster object.

## Context

Those labels are set when running commands such as :
```console
$ argocd cluster add --label foo=bar --annotation toto=baz ...
```

Currently, the only way I know of to view them is to use the argo-cd CLI, with a command like :
```console
$ argocd cluster list --output yaml
```

Those labels can be used by an ApplicationSet to generate a set of Application objects, e.g. one per cluster whose label match a certain value.

# Motivation

It would allow one to investigate from the UI, currently if someone is sent a link to an argo-cd instance but doesn't have a computer with the argo-cd CLI setup, they cannot verify the existing values of those cluster labels.

The consequence is that we may see an ApplicationSet generating 10 applications, while 20 clusters are defined, and have no way to check why some clusters are ignored.

This feature would allow to answer questions such as :
 * Did the person that added the cluster make a typo ?
 * What are the possible values ?
 * What cluster(s) would be impacted if I were to modify the selector of a given ApplicationSet ?

# Proposal

I took a look at the UI source code, the page in question is :
[src/app/settings/components/cluster-details/cluster-details.tsx](https://github.com/argoproj/argo-cd/blob/master/ui/src/app/settings/components/cluster-details/cluster-details.tsx)
but I wasn't able to understand where the cluster object was created and what it contained exactly (and when it was instantiated relatively to when the webpage is rendered), I'm guessing it doesn't have access the metadata in question.

So adding this feature would likely involve :
 1. Adding code that loads the argo-cd cluster object's metadata into the type-script cluster object
 2. Displaying this info in the "Cluster Details" page

Not being a web developer, it is hard for me to estimate the feasibility and complexity of this proposal.

Any feedback is welcome,
Thank you for your time.

**Comment by blakepettersson** [comment]
This should be fixed with #9707 I think?

**Comment by agagniere** [comment]
@blakepettersson  You are absolutely right, it is exactly that.
